app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -40,17 +42,26 @@ echo "#"
 
 <% if plugin_present?('katello') -%>
 if [ x$ID = xrhel ] || [ x$ID = xcentos ]; then
-    CONSUMER_RPM=$(mktemp --suffix .rpm)
+    register_katello_host(){
+        UUID=$(subscription-manager identity | head -1 | awk '{print $3}')
+        curl --request POST "<%= foreman_server_url %>/register?uuid=$UUID" \
+            <%= headers.join(' ') %> \
+            -d '{ "host": { "organization_id": "<%= @organization.id -%>"
+            <%= ", \"location_id\": \"#{@location.id}\"" if @location -%>
+            <%= ", \"hostgroup_id\": \"#{@hostgroup.id}\"" if @hostgroup -%>
+            }}'
+    }
 
-    curl --insecure --output $CONSUMER_RPM <%= subscription_manager_configuration_url %>
+    CONSUMER_RPM=$(mktemp --suffix .rpm)
+    curl --output $CONSUMER_RPM <%= subscription_manager_configuration_url %>
     yum localinstall $CONSUMER_RPM -y
-    yum install subscription-manager -y
     subscription-manager register --org=<%= @organization.label %> --activationkey="sample-key"
-
     rm -f $CONSUMER_RPM
```

Comment:
Nit: I would move this to just below `yum localinstall` to group that part.

In recent yum versions you can also use `yum install -y <%= subscription_manager_configuration_url %>` but I'm uncertain about EL6 so I'm fine with keeping the current `localinstall` method.